mlflow/models/evaluation/base.py

```diff
@@ -1096,7 +1096,7 @@ def evaluate(
     model: str,
     data,
     *,
-    model_type: str,
+    model_type: Optional[str] = None,
```

Comment:
Btw can we throw an exception that when model_type is blank, users need to specify extra_metrics or else nothing would be logged?
And maybe link to the docs on how to add them?